backend/src/hatchling/builders/utils.py

```diff
@@ -24,7 +24,7 @@ def safe_walk(path: str) -> Iterable[tuple[str, list[str], list[str]]]:
         stat = os.stat(root)
         identifier = stat.st_dev, stat.st_ino
         if identifier in seen:
-            del dirs[:]
+            dirs.clear()
```

Comment:
You might find this one questionable, since `del list[:]` is a widely used Python idiom. But then `list.clear()` is explicit, even for those with less exposure to Python.